### ISSUE TITLE
Build a single subdistrict boundary file

### ISSUE THREAD

**Issue by planemad** [justinelliotmeyers/INDIA_SOI_2022#1]
Cleaning the data upto the subdistrict level will be useful to work with the village shapes and perform other data joins. There are several gaps even at the subdistrict level which probably need to be addressed first.

**Missing coverage**

- Arunachal
- Dadra and Nagar Haveli and Daman and Diu
- Gujarat (can be built from villages)
- Jammu and Kashmir
- Ladakh
- Lakshadweep
- Maharashtra (can be built from villages)
- Meghalaya
- Sikkim
- Tripura

<img width="1003" alt="Screen Shot 2022-09-20 at 12 44 52 PM" src="https://user-images.githubusercontent.com/126868/191316660-9575c894-4a47-43bf-a35c-ccc8698c87ea.png">

**Outdated coverage**

Cursory glance shows that these subdistricts maybe not be accurate or incomplete
- Bihar
- Odisha
- West Bengal

<img width="422" alt="Screen Shot 2022-09-20 at 12 47 28 PM" src="https://user-images.githubusercontent.com/126868/191318917-fba14120-e195-4f25-84f8-c780ea8ed7c9.png">

**Comment by justinelliotmeyers** [comment]
So we should merge sub districts together from sub district files. Clean state name, district name, and sub district name. Does that sound like a plan? Should we merge these from the subdistricts or the villages? I'm thinking subdistricts but see value in doing from villages up.

**Comment by upperwal** [comment]
And maybe also map it to LGD. Let me know how I can help.

**Comment by justinelliotmeyers** [comment]
Merged file in Readme:
https://drive.google.com/drive/folders/1MOlQQC41_q1i0nHjf9p2XYLWNOkGQ_o4
I started work on this. Merged all subdistricts together for 25 states. Added the following fields:
CS - Clean State
CD - Clean District
CT - Clean Tehsil
CSLGD - Clean State LGD
RANDOM_ID - ID placeholder until we get SOI IDs or complete all Subdistrict LGDs

Added all state names and LGD codes. Had to fix Chhattisgarh and Uttar Pradesh to correct spelling.

**Comment by pratapvardhan** [comment]
Just added subdistricts metadata to [gsheets](https://docs.google.com/spreadsheets/d/1zu0-GMtpFuZ7XkxdfjQClWfIOGnj0GCiF-W2lL0rMsE/edit#gid=571215633) here, if it's easier to open it up for others to comment.

**Comment by justinelliotmeyers** [comment]
So far I only cleaned the CS state name. The districts and subdistricts all need qc/ cleaning. I think a good way to confirm them being cleaned is by populating the LGD codes. So if a districts LGD field is populated, then we know the name is stable and finished. Same with Tehsil

**Comment by pratapvardhan** [comment]
Should then first do this exercise on a single districts file and move here?

**Comment by justinelliotmeyers** [comment]
@pratapvardhan lets finish this file first. Then we can either do all districts where we dont have subdistrict data or bring in dissolved villages to substitute the missing subdistricts (Maharashtra / Gujarat)

**Comment by justinelliotmeyers** [comment]
If we can connect with the SOI and find out if they plan to release admins for missing states, that would be helpful, but not necessary right away. If not we may be able to bring in block data from the PMGSY project. But lets try to stick with one source for now.

**Comment by pratapvardhan** [comment]
@justinelliotmeyers I meant, isn't adding LGD codes in a downstream way more efficient. First fix district codes, then it's easier to match subdistricts within the district and then move to village codes? or am I missing something?

**Comment by justinelliotmeyers** [comment]
yeah, whatever way you want to work it or do it, jump in! Sometimes I do things backwards!>!? I did states first, and yes was planning on districts, then subdistricts. Wherever you can help and clean is appreciated!

**Comment by planemad** [comment]
Have added a sheet with the subdistricts from Gujarat and Maharashtra using the village shapes https://docs.google.com/spreadsheets/d/1zu0-GMtpFuZ7XkxdfjQClWfIOGnj0GCiF-W2lL0rMsE/edit#gid=499730654

**Comment by abhasdudeja** [comment]
@planemad @justinelliotmeyers I have recently cleaned the GJ village files, will request a pull in a while, you guys can use those files for GJ

**Comment by abhasdudeja** [comment]
> Cleaning the data upto the subdistrict level will be useful to work with the village shapes and perform other data joins. There are several gaps even at the subdistrict level which probably need to be addressed first.
> 
> **Missing coverage**
> 
> * Arunachal
> * Dadra and Nagar Haveli and Daman and Diu
> * Gujarat (can be built from villages)
> * Jammu and Kashmir
> * Ladakh
> * Lakshadweep
> * Maharashtra (can be built from villages)
> * Meghalaya
> * Sikkim
> * Tripura
> 
Regarding the Missing UT of DNH&DD.
One can import these PDFs to CAD and export the boundaries to Shapefiles -> Edit & Spatially Adjust.

Village names are also given in the PDFs

[DNH Plan](http://pdadnh.nic.in/Doc/PlanofDNH.pdf)
[Detailed Plan Part-2](http://pdadnh.nic.in/Doc/ODP_P2.pdf)
[Detailed Plan Part-1](http://pdadnh.nic.in/Doc/ODP_P1.pdf)
